### ISSUE TITLE
grid_points improvements

### ISSUE THREAD

**Issue by ryanlevy** [JoeyT1994/ITensorNumericalAnalysis.jl#27]
For certain values, `grid_points` will not get close to the value of 1.
For example, 50 grid points on L=6 will result with a max of 0.796875

![image](https://github.com/JoeyT1994/ITensorNumericalAnalysis.jl/assets/7354063/454e6a17-7b7a-4265-8c9a-1be4219e11fb)

Also a wishlist, it would nice to enforce a certain point (like I want a grid but make sure 0.5 is there)

**Comment by JoeyT1994** [comment]
Yeah I think the `grid_points` functionality should be massively fleshed out

To add to the wishlist we should also be able to sample randomly from the grid. For instance we could overload `rand` so that
`Random.rand(s::IndexMap)` randomly chooses a point on the d-dimensional grid defined by `IndexMap`
